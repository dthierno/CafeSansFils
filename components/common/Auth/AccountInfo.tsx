import { Image, Text, View, StyleSheet } from "react-native";

import COLORS from "@/constants/Colors";
import SPACING from "@/constants/Spacing";
import TYPOGRAPHY from "@/constants/Typography";

type AccountInfoProps = {
  title?: string;
  profileName: string;
  profilePicture: any;
};

export default function AccountInfo({
  title = "Bonjour et Bienvenue",
  profileName,
  profilePicture,
}: AccountInfoProps) {
  return (
    <View style={styles.accountContainer}>
      <Image source={profilePicture} style={styles.profilePicture} testID="header-account-image"/>
      <View>
        <Text style={[styles.welcomeText, TYPOGRAPHY.body.normal.base]}>
          {title}
        </Text>
        <Text style={[styles.userFullName, TYPOGRAPHY.heading.small.bold]}>
          {profileName}
        </Text>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  accountContainer: {
    flexDirection: "row",
    alignItems: "center",
    gap: SPACING.sm,
  },
  profilePicture: {
    width: SPACING["8xl"],
    height: SPACING["8xl"],
    borderRadius: "100%",
    borderWidth: 4,
    borderColor: "rgba(0, 0, 0, 0.1)",
  },
  welcomeText: {},
  userFullName: {
    color: COLORS.black,
  },
});
