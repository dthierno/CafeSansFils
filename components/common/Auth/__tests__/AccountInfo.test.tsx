import React from "react";
import { render } from "@testing-library/react-native";
import AccountInfo from "@/components/common/Auth/AccountInfo";

describe("AccountInfo", () => {
  const mockProps = {
    title: "Welcome Back",
    profileName: "John Doe",
    profilePicture: 'link/to/profile/picture',
  };

  it("renders the profile picture", () => {
    const { getByTestId } = render(<AccountInfo {...mockProps} />);
    const profilePicture = getByTestId("header-account-image");
    expect(profilePicture.props.source).toEqual(mockProps.profilePicture);
  });

  it("renders the title", () => {
    const { getByText } = render(<AccountInfo {...mockProps} />);
    expect(getByText(mockProps.title)).toBeTruthy();
  });

  it("renders the user's full name", () => {
    const { getByText } = render(<AccountInfo {...mockProps} />);
    expect(getByText(mockProps.profileName)).toBeTruthy();
  });

    it("matches snapshot", () => {
        const { toJSON } = render(<AccountInfo {...mockProps} />);
        expect(toJSON()).toMatchSnapshot();
    });
});
