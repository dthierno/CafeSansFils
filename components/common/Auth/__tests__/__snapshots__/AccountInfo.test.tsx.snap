// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountInfo matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "gap": 12,
    }
  }
>
  <Image
    source="link/to/profile/picture"
    style={
      {
        "borderColor": "rgba(0, 0, 0, 0.1)",
        "borderRadius": "100%",
        "borderWidth": 4,
        "height": 52,
        "width": 52,
      }
    }
    testID="header-account-image"
  />
  <View>
    <Text
      style={
        [
          {},
          {
            "fontFamily": "Inter-Regular",
            "fontSize": 12,
          },
        ]
      }
    >
      Welcome Back
    </Text>
    <Text
      style={
        [
          {
            "color": "#000000",
          },
          {
            "fontFamily": "Poppins-Bold",
            "fontSize": 18,
          },
        ]
      }
    >
      John Doe
    </Text>
  </View>
</View>
`;
